docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+## 2. Setting up
 
+### 2.1. Prerequisites
+1. JDK `11`.
+2. IntelliJ IDE.
 
-## Product Scope
-### Target user profile
+### 2.2. Setting up the project in your computer
+1. Fork this repository, and clone the fork repository to your computer.
+2. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle  
+   1. Click `Configure` > `Structure for New Projects` and then `Project Settings` > `Project` > `Project SDK`.
+   2. If `JDK 11` is listed in the drop down, select it. Otherwise, click `New…` and select the directory where you installed `JDK 11`.
+   3. Click `OK`.
+4. Click `Import Project`.
+5. Locate the `build.gradle` file and select it. Click `OK`.
+6. Click `Open as Project`.
+7. Click `OK` to accept the default settings if prompted. 
 
-{Describe the target user profile}
+## 3. Design
+This section provides a high level overview of our application, Kitchen Helper.
+### 3.1. Architecture
 
-### Value proposition
+![Architecture](images/KitchenHelperMain.png)
+The image above explains the design of the application, Kitchen Helper. 
 
-{Describe the value proposition: what problem does it solve?}
+The main driver of the application is `Main: Kitchen Helper`. It is responsible for mainly two phases:
+- At application launch
+    - This class will initialise the components in the correct sequence and is in charge of connecting them with each other.
+- At shut down
+    - This class will invoke cleanup method for the components when necessary.
+    
+In addition to that, the architecture of Kitchen Helper is broken down into seven classes, mainly the following: 
+- `Ui`: This class mainly handles the UI of the application.
+- `Parser`: This class mainly handles the parsing and handling of user commands.
+- `Command`: This class handles the type of command.
+- `Ingredient`: This class manages the data of data type ingredient in memory.
+- `Chore`: This class manages the data of data type chore in memory.
+- `Recipe`: This class manages the data of data type recipe in memory.
+- `Storage`: This class reads data from and writes data back into a text file for future uses.
 
-## User Stories
+### 3.2. Ui Component
+![Ui Component](images/UI_Component.png)
+
+API: `Ui.java`
+ 
+The `Ui` component is a singleton class where all interaction will be made through this component
+ 
+The `Ui` component,
+
+* Executes user commands using the command component
+* Listens for changes and outputs messages from the Command component
+
+### 3.3. Logic Component
+
+![Logic Component](images/logic.png)
+
+1. `Kitchen Helper` uses `Parser` class to parse the user command.
+2. This results in a command object return back which is executed by `Kitchen Helper`.
```

Comment:
The Kitchen Helper, Parser, CommandResult and Ui are in another special formatting but the Command Object is not, myabe you can work on how to maintain the consistency of your formatting in the developer guide.